### ISSUE TITLE
ERROR: Maximum call stack size exceeded when using json/nested

### ISSUE THREAD

**Issue by lukasoppermann** [amzn/style-dictionary#464]
Hey, I am getting the following error when trying to transform my json file. I am a bit at a loss what to do about it. 😅

Please let me know how I can provide some more helpful information.

> /Users/lukasoppermann/Repos/design-token-transformer/node_modules/style-dictionary/lib/common/formats.js:75
  if (obj.hasOwnProperty('value')) {
          ^

RangeError: Maximum call stack size exceeded
    at String.hasOwnProperty (<anonymous>)
    at minifyDictionary (/Users/lukasoppermann/Repos/design-token-transformer/node_modules/style-dictionary/lib/common/formats.js:75:11)
    at minifyDictionary (/Users/lukasoppermann/Repos/design-token-transformer/node_modules/style-dictionary/lib/common/formats.js:80:23)
    at minifyDictionary (/Users/lukasoppermann/Repos/design-token-transformer/node_modules/style-dictionary/lib/common/formats.js:80:23)
    at minifyDictionary (/Users/lukasoppermann/Repos/design-token-transformer/node_modules/style-dictionary/lib/common/formats.js:80:23)
    at minifyDictionary (/Users/lukasoppermann/Repos/design-token-transformer/node_modules/style-dictionary/lib/common/formats.js:80:23)
    at minifyDictionary (/Users/lukasoppermann/Repos/design-token-transformer/node_modules/style-dictionary/lib/common/formats.js:80:23)
    at minifyDictionary (/Users/lukasoppermann/Repos/design-token-transformer/node_modules/style-dictionary/lib/common/formats.js:80:23)
    at minifyDictionary (/Users/lukasoppermann/Repos/design-token-transformer/node_modules/style-dictionary/lib/common/formats.js:80:23)
    at minifyDictionary (/Users/lukasoppermann/Repos/design-token-transformer/node_modules/style-dictionary/lib/common/formats.js:80:23)

**Comment by chazzmoney** [comment]
@lukasoppermann 
Looks like there may be an infinite loop. Are you running vanilla SD or are you using some of the custom work you've built?

What would enable us to help:
1. The config you are using
2. Any custom code you've built
3. The source dictionaries

**Comment by lukasoppermann** [comment]
I think I have found what causes it, in the json below the `comment` key on line 3 (as a child of the h3) causes the issue. Removing this makes it work. However having additional comments & meta info on a group is important for documenting the system.

```json5
"body": {
    "h3": {
      "comment": "Comment for text style",
      "fontSize": {
        "value": 20,
        "type": "number",
        "unit": "pixel"
      },
      "textDecoration": {
        "value": "none",
        "type": "string"
      }, //…
```

**Comment by dbanksdesign** [comment]
Found the issue and opened a PR for it: #465

**Comment by chazzmoney** [comment]
@dbanksdesign while it looks like we have a solution for the issue now, it still stands that there may be another data state in which an infinite loop could occur here. Do we want to try to catch that?

**Comment by dbanksdesign** [comment]
Good call. I'll try to think of other cases that would result in an infinite loop...

**Comment by dbanksdesign** [comment]
This is now merged and released in v2.10.2!

**Comment by lukasoppermann** [comment]
Thank you!
